src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -121,6 +122,8 @@ def __getitem__(self, item):
             return self.secrets[item]
         if item in self.configs:
             return self.configs[item]
+        # raise UserErrorException(error=KeyError(f"Key {item!r} not found in connection {self.name!r}."))
+        # Cant't raise UserErrorException due to the code exit(1) of promptflow._cli._utils.py line 368.
```

Comment:
I think we can still raise UserErrorException, since the CLI's exit(1) will happen after we log the exception type?

Reply:
I haven't made this modification yet because it will cause a lot of test errors. The reason is that it exited the program directly instead of throwing an exception for the upstream to receive, so I am also worried that there may be many unexpected problems if I make the modification.